plugins/woocommerce/src/Blocks/BlockTypes/Checkout.php

```diff
@@ -159,6 +163,14 @@ protected function get_block_type_script( $key = null ) {
 			$dependencies[] = 'zxcvbn-async';
 		}
 
+		$checkout_fields = Package::container()->get( CheckoutFields::class );
+		$checkout_schema = Package::container()->get( CheckoutFieldsSchema::class );
+
+		// Load schema parser asynchronously if we need it.
+		if ( $checkout_schema->has_valid_schema( $checkout_fields->get_additional_fields() ) && Features::is_enabled( 'experimental-blocks' ) ) {
+			$dependencies[] = 'wc-schema-parser';
+		}
+
```

Comment:
You should do the experimental block check first IMO to prevent the classes being initialised.

```suggestion
		if ( Features::is_enabled( 'experimental-blocks' ) ) {
				$checkout_fields = Package::container()->get( CheckoutFields::class );
				$checkout_schema = Package::container()->get( CheckoutFieldsSchema::class );

				// Load schema parser asynchronously if we need it.
				if ( $checkout_schema->has_valid_schema( $checkout_fields->get_additional_fields() ) ) {
					$dependencies[] = 'wc-schema-parser';
				}
		}

```